lib/pbench/server/api/resources/users_api.py

```diff
@@ -0,0 +1,534 @@
+
+        This requires a JSON data with required user metadata fields
+        {
+            "username": "username",
+            "password": "password",
+        }
+
+        Required headers include
+
+            Content-Type:   application/json
+            Accept:         application/json
+
+        :return: JSON Payload
+        if we succeed to decrypt the password hash, the returned response_object will include the auth_token
+            response_object = {
+                    "status": "success", # will not present if failed
+                    "message": "Successfully logged in."/"failure message",
+                    "auth_token": "<authorization_token>", # Will not present if failed
+                }
+        """
+        # get the post data
+        post_data = request.get_json()
+        if not post_data:
+            self.logger.warning("Invalid json object: {}", request.url)
+            abort(400, message="Invalid json object in request")
+
+        username = post_data.get("username")
+        if not username:
+            self.logger.warning("Username not provided during the login process")
+            abort(400, message="Please provide a valid username")
+
+        password = post_data.get("password")
+        if not password:
+            self.logger.warning("Password not provided during the login process")
+            abort(400, message="Please provide a valid password")
+
+        try:
+            # fetch the user data
+            user = User.query(username=username)
+        except Exception:
+            self.logger.exception("Exception occurred during user login")
+            abort(500, message="INTERNAL ERROR")
+
+        if not user:
+            self.logger.warning(
+                "No user found in the db for Username: {} while login", username
+            )
+            abort(403, message="No such user, please register first")
+
+        # Validate the password
+        if not check_password_hash(user.password, password):
+            self.logger.warning("Wrong password for user: {} while login", username)
+            abort(401, message="Bad login")
+
+        try:
+            auth_token = self.auth.encode_auth_token(
+                self.token_expire_duration, user.id
+            )
+        except (
+            jwt.InvalidIssuer,
+            jwt.InvalidIssuedAtError,
+            jwt.InvalidAlgorithmError,
+            jwt.PyJWTError,
+        ):
+            self.logger.error(
+                "Could not encode the JWT auth token for user: {} while login", username
+            )
+            abort(
+                500, message="INTERNAL ERROR",
+            )
+
+        # check if the user is already logged in, in case the request has a an authorization header included
+        # We would still proceed to login and return a new auth token to the user
+        if user == self.auth.token_auth.current_user():
+            self.logger.warning("User already logged in and trying to re-login")
```

Comment:
Idk, you think this should be an info message instead of warning in our logs?

Reply:
I don't know either, which is why I was asking.  :-D

Since we do allow the user to re-log-in (at least, if they don't do it tooooo often...), it doesn't seem like an actual occurrence of it should warrant a warning.  You seem to think that it is interesting enough to warrant logging it, but it doesn't seem likely to be a problem (if it were, it seems like we would have more code around it...).

Reply:
yeah that also make sense